src/gui/plugins/transform_control/TransformControl.cc

```diff
@@ -128,9 +250,37 @@ void TransformControl::OnMode(const QString &_mode)
       ignerr << "Error setting transform mode" << std::endl;
   };
 
-  ignition::msgs::StringMsg req;
-  req.set_data(_mode.toStdString());
-  this->dataPtr->node.Request(this->dataPtr->service, req, cb);
+  auto modeStr = _mode.toStdString();
+
+  // Legacy behaviour: send request to GzScene3D
+  if (this->dataPtr->legacy)
+  {
+    ignition::msgs::StringMsg req;
+    req.set_data(modeStr);
+    this->dataPtr->node.Request(this->dataPtr->service, req, cb);
+  }
+  // New behaviour: handle the transform control locally
+  else
+  {
+    std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+    if (modeStr == "select")
+      this->dataPtr->transformMode = rendering::TransformMode::TM_NONE;
+    else if (modeStr == "translate")
+      this->dataPtr->transformMode = rendering::TransformMode::TM_TRANSLATION;
+    else if (modeStr == "rotate")
+      this->dataPtr->transformMode = rendering::TransformMode::TM_ROTATION;
+    else if (modeStr == "scale")
+      this->dataPtr->transformMode = rendering::TransformMode::TM_SCALE;
+    else
+      ignerr << "Unknown transform mode: [" << modeStr << "]" << std::endl;
+
+    ignition::gazebo::gui::events::TransformControlMode transformControlMode(
+      this->dataPtr->transformMode);
```

Comment:
As far as I remember, Gazebo `GuiEvent.hh` does not depend on `ign-rendering`, I can't not include `#include "ignition/rendering/TransformType.hh"` to use the rendering emuns, then I noticed that checking if the transform control mode is active was enough. I can use a better name for the event, for example `TransformControlModeActive`.

Reply:
Thanks, I think the new name is good.

But we're still converting an `enum` to a `bool` and counting on the fact that `TA_NONE` happens to be zero. I think this is fine for now, but something to keep in mind in case the enums change.